text/0102-rawkv-br.md

```diff
@@ -0,0 +1,92 @@
+
+<img src="../media/tikv-br.png" alt="RawKV BR" width="60%"/>
+
+## Detailed design
+
+### Backup
+
+The main process of the backup is as following:  
+1. TiKV-BR client launches the backup process with user's command with given range and storage.
+2. TiKV-BR client gets all stored from pd and send [`BackupRequest`] to every TiKV store.
+3. TiKV nodes receive the `BackupRequest` and scan all regions whose range has intersection with the backup range and whose leader is the current store.
+4. TiKV nodes write the scanned raw key-value entries to storage in RocksDB SST format.
+5. TiKV nodes send [`BackupResponse`] to TiKV-BR client.
+5. TiKV-BR client writes [`BackupMeta`] to storage.
+
+From above process, we can see that the backup process is distributed and paralleled. This can speedup the backup with increment of TiKV nodes and regions. In a large scale TiKV cluster, the backup bottleneck usually be the network bandwidth or the storage iops. However, backup scan consumes much CPU, which may affect the performance of the frontend. User can specify the `ratelimit` to limit the backup speed to decrease the performance impact on the frontend.  
+
+RawKV Backup supports [`api-version`] conversion from V1 to V2 with config `--dst-api-version=V2`. Then restore the backup files to API V2 `TiKV` cluster. This is mainly used to upgrade from API V1 cluster to API V2 cluster. In this scenario, TiKV nodes do the conversion during scan and there's no need to do rewrite during restoration.  
+
+RawKV backup returns the `backup-ts` in [API V2] cluster, which can be used as `start-ts` of `TiKV-CDC` when creating replication tasks.  
+
+### Restoration
+
+The main process of Restoration is as following:  
+1. TiKV-BR client launched the restoration process with user's command with given range and storage.
+2. TiKV-BR client requests TiKV cluster switch to `Import` mode.
+3. TiKV-BR client reads the [`BackupMeta`] from storage and get the SST files whose range has intersection with the backup range.
+4. TiKV-BR client sends [SplitRangeRequest] to TiKV store to pre-split regions with the SST files's range.
+5. TiKV-BR client sends [DownloadRequest] to the TiKV stores which the SST file's range belongs to. TiKV nodes download SST files from storage.
+6. TiKV-BR client sends [IngestRequest] to every affected region's leader node to ingest SST files to RocksDB with raft procedure.
+
+In `Import` mode, following changes will occur:
+- Let `PD` reduces region schedule frequency to speedup restoration.
+- TiKV stores change `RocksDB`'s configuration, for example, increase the `level0_stop_writes_trigger` and `level0_slowdown_writes_trigger`, decrease background compaction tasks, which can speedup ingest process.
+
+With the `SplitRegion` step, we can balance the data between TiKV stores.  
```

Comment:
We know the data size and distributin, so pre-split regions before import, which has very little costs.

Otherwise, splitting & rebalance after import needs data movement, which would be very expensive.